test/specs/validate-address.spec.js

```diff
@@ -694,9 +704,101 @@ describe("validateAddress()", () => {
     }
   });
 
+  it("Validates an address with name,company,and phone", async function () {
+    const shipengine = new ShipEngine({ apiKey, baseURL });
+
+    const addressToValidate = {
+      country: "US",
+      name: "Bruce Wayne",
+      phone: "1234567891",
+      company: "ShipEngine",
+      street: ["4 Jersey St", "address-with-personal-info"],
+      cityLocality: "Boston",
+      stateProvince: "MA",
+      postalCode: "02215",
+    };
+
+    const expectedNormalizedAddress = {
+      country: "US",
+      street: ["4 JERSEY ST"],
+      cityLocality: "BOSTON",
+      stateProvince: "MA",
+      postalCode: "02215",
+      name: "BRUCE WAYNE",
+      company: "SHIPENGINE",
+      phone: "1234567891",
+    };
+
+    const response = await shipengine.validateAddress(addressToValidate);
+
+    const { normalizedAddress, isValid } = response;
+
+    // The isValid flag is true
+    expect(isValid).to.be.a("boolean").and.to.equal(true);
+
+    // The normalized address is populated and matches the expected normalized address
+    assertAddressEquals(normalizedAddress, expectedNormalizedAddress);
+
+    // The isResidential flag on the normalized address is true
+    expect(normalizedAddress.isResidential)
+      .to.be.a("boolean")
+      .and.to.equal(false);
+
+    // There are no warning or error messages
+    assertNoWarningsOrErrorMessages(response);
+
+    // It should have a normalized address with the correct shape
+    assertNormalizedAddressFormat(normalizedAddress);
+  });
+
+  it("Validates an address with NO name,company,and phone", async function () {
+    const shipengine = new ShipEngine({ apiKey, baseURL });
+
+    const addressToValidate = {
+      country: "US",
+      street: ["4 Jersey St", "validate-residential-address"],
+      cityLocality: "Boston",
+      stateProvince: "MA",
+      postalCode: "02215",
+    };
+
+    const expectedNormalizedAddress = {
+      country: "US",
+      street: ["4 JERSEY ST"],
+      cityLocality: "BOSTON",
+      stateProvince: "MA",
+      postalCode: "02215",
+      name: "",
+      company: "",
+      phone: "",
+    };
+
+    const response = await shipengine.validateAddress(addressToValidate);
+
+    const { normalizedAddress, isValid } = response;
+
+    // The isValid flag is true
+    expect(isValid).to.be.a("boolean").and.to.equal(true);
+
+    // The normalized address is populated and matches the expected normalized address
+    assertAddressEquals(normalizedAddress, expectedNormalizedAddress);
+
+    // The isResidential flag on the normalized address is true
+    expect(normalizedAddress.isResidential)
+      .to.be.a("boolean")
+      .and.to.equal(true);
```

Comment:
Why test for this separately? Why not just include it in the `expectedNormalizedAddress` object?

Reply:
Just to be explicit. I can definitely add it there as well.